plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerSession.kt

```diff
@@ -154,7 +154,9 @@ class CodeModernizerSession(
         } catch (e: Exception) {
             state.putJobHistory(sessionContext, TransformationStatus.FAILED)
             state.currentJobStatus = TransformationStatus.FAILED
```

Comment:
I'm trying to discern if we can modify `e: Exception` to some sort of AWS SDK error object like `AwsServiceException` -> https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/error-handling.html

You should be able to do some error handling for this type of error object and directly report this on a 403 with:
```
        } catch(e: AwsServiceException) {
            if (e.statusCode() === 403 && e.awsErrorDetails().serviceName() === "S3") {
                // do something
            }
        }
```

Reply:
I'm not exactly sure about the service name string, but since `startJob` and `s3Upload` are in the same try/catch, we could possibly use this to discern the difference between the two.

Reply:
@rli Is the `AwsServiceException` the right way to capture AWS SDK specific issues in Kotlin?

Reply:
`S3Exception` / `CodeWhisperer[Streaming]Exception`?
avoid introspecting status codes

Reply:
Shall I go with `S3Exception` or `AwsServiceException ` to capture the upload failure to S3.